src/network/websockets/server/index.ts

```diff
@@ -44,43 +44,65 @@ export function runWebsocketServer(params: Params) {
 
     const handlers: { [action: string]: (msg: any) => void } = {
       ping: () => send({ action: "pong" }),
-      push: msg => {
-        const { kind, id, lastSeenRevision, value } = msg;
-        onChangeData({
-          kind,
-          lastSeenRevision,
-          id,
-          value: serialization.decode(value),
-          send: v => events.emit("change", { kind, id, value: v })
+      auth: msg => authenticate(msg),
```

Comment:
I don’t think this restructuring is very beneficial. We had three functions with function names and now we have three functions with function names. 🤷‍♂ 

More importantly, it is harmful for the PR. Instead of the diff showing the actually added functionality, this shows as if everything was rewritten. I don’t know which lines have changed and which stayed the same, so reviewing this becomes much harder. If you do the refactors in a separate PR, or at least in a separate commit, I can focus on the functional changes and evaluate them so much easier! 🎆 

```tsx
// Before PR
const handlers = {
  auth: msg => {
    //
  },
  push: msg => {
    //
  },
  subscribe: msg => {
    //
  }
}

// After PR
const handlers = { auth, push, subscribe };
function auth(msg) {
  //
},
function push(msg) {
  //
},
function subscribe(msg) {
  //
}
```